packages/grafana-data/src/dataframe/FieldCache.ts

```diff
@@ -58,11 +58,9 @@ export class FieldCache {
   }
 
   getFirstFieldOfType(type: FieldType): FieldWithIndex | undefined {
-    const arr = this.fieldByType[type];
-    if (arr && arr.length > 0) {
-      return arr[0];
-    }
-    return undefined;
+    const fields = this.fieldByType[type];
+    const firstField = fields.find(field => !(field.config.custom && field.config.custom['Hidden']));
```

Comment:
Maybe not the best place to put this logic, but I needed a way to prevent fields I'd marked as "Hidden" - fields whose data are potentially needed later to get surrounding context, but that the user doesn't need to know about - from being retrieved.

Reply:
I think this could make sense but maybe this is more a presentation problem? To me this would make more sense to do in Explore or Table panel where we would filter any "hidden" fields just before showing them but for general low level accessors over dataframe I would not expect it to be filtered in this way.

Reply:
I agree, this was more of a quick and dirty way to get things working. I'll take a stab at a better design